test/jdk/javax/swing/JTable/CellFocusRingTest.java

```diff
@@ -0,0 +1,153 @@
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 7124282
+ * @key headful
+ * @requires (os.family == "mac")
+ * @summary Checks whether Table & List's focus ring color is more prominent
+ * in comparison to original focus ring (represented by 'CellFocus.color'
+ * property in Aqua LAF UIDefaults) for MacOS accent colors.
+ * @run main CellFocusRingTest
+ */
+
+import java.awt.Color;
+import javax.swing.plaf.BorderUIResource.LineBorderUIResource;
+import javax.swing.UIManager;
+
+public class CellFocusRingTest {
+
+    private static StringBuffer errorLog = new StringBuffer();
+
+    public static void main(String[] args) {
+        try {
+            UIManager.setLookAndFeel("com.apple.laf.AquaLookAndFeel");
+        } catch (Exception e) {
+            throw new RuntimeException("Unsupported Look&Feel Class");
+        }
+
+        System.out.println("Test for Table");
+        testFocusRingColor("Table");
+        System.out.println("--------------------------------");
+
+        System.out.println("Test for List");
+        testFocusRingColor("List");
+        System.out.println("--------------------------------");
+
+        if (errorLog.isEmpty()) {
+            System.out.println("Test passed !!");
+        }
+        else {
+            throw new RuntimeException("Following cases failed.\n"+ errorLog);
+        }
+    }
+
+    private static void testFocusRingColor(String prefix) {
+        float[] bckRGB = new float[3];
+        float[] oldCellRingRGB = new float[3];
+        float[] newCellRingRGB = new float[3];
+
+        Color selectionBck = null;
+        Color originalRingColor = null;
+        Color newRingColor = null;
+
+        // saturation threshold for grayish colors
+        float satGrayScale = 0.10f;
+
+        if (UIManager.getDefaults().get(prefix + ".selectionBackground") != null
+                && UIManager.getDefaults().get(prefix + ".selectionBackground")
+                instanceof Color) {
+            selectionBck = (Color) UIManager.getDefaults()
+                    .get(prefix + ".selectionBackground");
+        }
+
+        if (UIManager.getDefaults().get("CellFocus.color") != null
```

Comment:
```
        if (UIManager.getDefaults().get("CellFocus.color") != null
                && UIManager.getDefaults().get("CellFocus.color")
                instanceof Color)
```
can be changed into 
```
        if (UIManager.getDefaults().get("CellFocus.color")
                instanceof Color)
```
Since if the object is null then `instanceof` returns false.

Reply:
@TejeshR13 The second half of the condition is executed only if  `"CellFocus.color" != null` (due to and conditional)